hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/AbstractSessionCredentialsProvider.java

```diff
@@ -154,21 +152,4 @@ public String toString() {
   public IOException getInitializationException() {
     return initializationException;
   }
-
-  /**
-   * A special set of null credentials which are not the anonymous class.
-   * This will be interpreted as "this provider has no credentials to offer",
-   * rather than an explicit error or anonymous access.
-   */
-  protected static final class NoCredentials implements AWSCredentials {
```

Comment:
hmm I guess this is probably not used anywhere as it didn't show up when I was doing my credentials work. But instead of removing, let's instead make it implement V2's AwsCredentials